sklearn/metrics/_classification.py

```diff
@@ -1889,7 +1893,6 @@ def precision_recall_fscore_support(
         weights = None
 
     if average is not None:
-        assert average != "binary" or precision.shape[0] == 1
```

Comment:
Why remove the assert statement?

Reply:
Oh this is because this seems to be unnecessary. 

@adrinjalali pointed this out in an earlier PR that we should either remove it or handle the error properly.

I decided to just do it in this PR as the main changes are minimal.

Reply:
Thanks for explaining. Fine for me to keep it here